SHARED_FILES_ANALYSIS.md

```diff
@@ -0,0 +1,192 @@
+11. TempFileUtilities.cs
+12. Tracing.cs ✅ **MOVED**
+13. VersionUtilities.cs ✅ **MOVED**
+
+**Status**: 6 files moved, 7 remaining
+
+#### Category 2: Harder to Move (In MSBuildTaskHost + Others)
+These 32 files are used in MSBuildTaskHost AND other projects. Moving these would require either:
+1. Making MSBuildTaskHost depend on Framework, OR
+2. Keeping duplicates (in Shared for MSBuildTaskHost, also in Framework)
+
+Files in this category include:
+- AssemblyNameComparer.cs
+- AssemblyNameExtension.cs
+- BinaryReaderExtensions.cs
+- BinaryWriterExtensions.cs
+- BuildEnvironmentHelper.cs
+- CommunicationsUtilities.cs
+- Constants.cs
+- CopyOnWriteDictionary.cs
+- EnvironmentUtilities.cs
+- ErrorUtilities.cs
+- EscapingUtilities.cs
+- ExceptionHandling.cs
+- FileUtilities.cs
+- FileUtilitiesRegex.cs
+- INodeEndpoint.cs
+- INodePacket.cs
+- INodePacketFactory.cs
+- INodePacketHandler.cs
+- InterningBinaryReader.cs
+- LogMessagePacketBase.cs
+- Modifiers.cs
+- NamedPipeUtil.cs
+- NodeBuildComplete.cs
+- NodePacketFactory.cs
+- NodeShutdown.cs
+- ReadOnlyEmptyCollection.cs
+- ReadOnlyEmptyDictionary.cs
+- ResourceUtilities.cs
+- TaskParameter.cs
+- TaskParameterTypeVerifier.cs
+- TranslatorHelpers.cs
+- XMakeAttributes.cs
+
+#### Category 3: MSBuildTaskHost Only (Should NOT Move)
+These 15 files are only used in MSBuildTaskHost and should remain in Shared:
+- AssemblyLoadInfo.cs
+- BufferedReadStream.cs
+- CollectionHelpers.cs
+- FileSystem/FileSystems.cs
+- FileSystem/IFileSystem.cs
+- IsExternalInit.cs
+- LoadedType.cs
+- NodeEndpointOutOfProcBase.cs
+- NodeEngineShutdownReason.cs
+- OutOfProcTaskHostTaskResult.cs
+- TaskEngineAssemblyResolver.cs
+- TaskHostConfiguration.cs
+- TaskHostTaskCancelled.cs
+- TaskHostTaskComplete.cs
+- TaskLoader.cs
+
+## Work Completed
+
+### Phase 1: Simple Files with No Dependencies ✅
+Successfully moved 6 files to Framework that had no complex dependencies:
+- CanonicalError.cs
+- FileDelegates.cs
+- Tracing.cs
+- VersionUtilities.cs
+- AssemblyFolders/Serialization/AssemblyFolderCollection.cs
+- AssemblyFolders/Serialization/AssemblyFolderItem.cs
+
+**Changes Made:**
+1. Copied files from `src/Shared/` to `src/Framework/`
+2. Removed Compile Include references from:
+   - Utilities/Microsoft.Build.Utilities.csproj
+   - Tasks/Microsoft.Build.Tasks.csproj
+   - Build/Microsoft.Build.csproj
+   - MSBuild/MSBuild.csproj
+   - MSBuild.UnitTests/Microsoft.Build.CommandLine.UnitTests.csproj
+3. Files auto-included in Framework by SDK (EnableDefaultItems not set to false)
+
+**Result:** Build succeeds ✅
+
+## Remaining Work
+
+### Category 1 Remaining Files (7 files)
+
+These files have dependencies that would need to be addressed:
+
+1. **AssemblyFoldersEx.cs**
+   - Depends on: Microsoft.Win32.Registry
+   - Requires adding package reference to Framework
+
+2. **AssemblyFoldersFromConfig.cs**
+   - Depends on: ErrorUtilities
+   - ErrorUtilities is in Category 2 (also needed by MSBuildTaskHost)
```

Comment:
that's not a problem, the good state is that shared files are compiled by framework and taskhost, not by taskhost and 2/3 other projects